pkg/controller.v1/tensorflow/tensorflow.go

```diff
@@ -107,11 +107,6 @@ func genClusterSpec(tfjob *tfv1.TFJob) (ClusterSpec, error) {
 	clusterSpec := make(ClusterSpec)
 
 	for rtype, spec := range tfjob.Spec.TFReplicaSpecs {
-		if rtype == tfv1.TFReplicaTypeEval {
-			// https://www.tensorflow.org/api_docs/python/tf/estimator/RunConfig
-			// evaluator is not part of training cluster
-			continue
-		}
```

Comment:
Does it work when we do not use estimator distribute strategy?

Reply:
This is just for the runconfig. If we are not using estimator then the runconfig will just contain the other replica types. See unit tests in the file above.

Reply:
Why remove it? Does this patch have a description of the problem it solves?
Thanks @gaocegege @richardsliu

Reply:
It is to add the evaluator in the cluster spec, I think.

Reply:
In tf2.4 PSS, if evaluator in the cluster spec, training job will hang in coordinator. Do we need to make some extra config in tf-operator? for example, exclude evaluator config for ps/worker.

Reply:
Do you mean it does not work with ParameterServerStrategy?

Reply:
I think we can open a new issue to keep track of the problem.

Reply:
yes. if evaluator not start rpc server, the coordinator will hang. [related linked issue](https://github.com/tensorflow/tensorflow/issues/46169).
But i just test it in our Yarn cluster not k8s, so I want to know how to solve it in tf-operator. Maybe tf-operator have similar problems.